doc/man3/SSL_shutdown.pod

```diff
@@ -21,161 +21,242 @@ SSL_shutdown, SSL_shutdown_ex - shut down a TLS/SSL or QUIC connection
+attempt is made to send a close_notify alert, regardless of whether the attempt
+was successful. The B<SSL_RECEIVED_SHUTDOWN> flag is set once a close_notify
+alert is received, which may occur during any call which processes incoming data
+from the network, such as L<SSL_read(3)> or SSL_shutdown(). These flags
+may be checked using L<SSL_get_shutdown(3)>.
+
+=head2 Fast Shutdown
+
+Alternatively, it is acceptable for an application to call SSL_shutdown() once
+(such that it returns 0) and then close the underlying connection without
+waiting for the peer's response. This allows for a more rapid shutdown process
+if the application does not wish to wait for the peer.
+
+This alternative "fast shutdown" approach should only be done if it is known
+that the peer will not send more data, otherwise there is a risk of an
+application exposing itself to a truncation attack. The full SSL_shutdown()
+process, in which both parties send close_notify alerts and SSL_shutdown()
+returns 1, provides a cryptographically authenticated indication of the end of a
+connection.
+
+This approach of a single SSL_shutdown() call without waiting is preferable to
+simply calling L<SSL_free(3)> or L<SSL_clear(3)> as calling SSL_shutdown()
+beforehand makes an SSL session eligible for subsequent reuse and notifies the
+peer of connection shutdown.
+
+The fast shutdown approach can only be used if there is no intention to reuse
+the underlying connection (e.g. a TCP connection) for further communication; in
+this case, the full shutdown process must be performed to ensure
+synchronisation.
+
+=head2 Effects on Session Reuse
+
+Calling SSL_shutdown() sets the SSL_SENT_SHUTDOWN flag (see
+L<SSL_set_shutdown(3)>), regardless of whether the transmission of the
+close_notify alert was successful or not. This makes the SSL session eligible
+for reuse; the SSL session is considered properly closed and can be reused for
+future connections.
+
+=head2 Quiet Shutdown
+
+SSL_shutdown() can be modified to set the connection to the "shutdown"
+state without actually sending a close_notify alert message; see
+L<SSL_CTX_set_quiet_shutdown(3)>. When "quiet shutdown" is enabled,
+SSL_shutdown() will always succeed and return 1 immediately.
+
+This is not standards-compliant behaviour. It should only be done when the
+application protocol in use enables the peer to ensure that all data has been
+received, such that it doesn't need to wait for a close_notify alert, otherwise
+application data may be truncated unexpectedly.
+
+=head2 Non-Compliant Peers
+
+There are SSL/TLS implementations that never send the required close_notify
+alert message but simply close the underlying transport (e.g. a TCP connection)
+instead. This will ordinarily result in an error being generated.
+
+If compatibility with such peers is desired, the option
+B<SSL_OP_IGNORE_UNEXPECTED_EOF> can be set. For more information, see
+L<SSL_CTX_set_options(3)>.
```

Comment:
The alternative is that you avoid calling SSL_read() after you know all data has been received, so that an EOF is not detected. The flag is handy for cases where you can't fix the code to avoid the unexpected EOF (because it's some library for instance.)